### PR TITLE
Split attribute visibility

### PR DESCRIPTION
**Pull request by dcbaker**

### CODE REVIEW

File: docs/markdown/Reference-tables.md [hunk 1]
```diff
@@ -182,12 +182,22 @@ which are supported by GCC, Clang, and other compilers.
...
+| visibility:protected†|
 | warning              |
 | warn_unused_result   |
 | weak                 |
 | weakreaf             |
 
+\* *Changed in 0.52.0* the "visibility" target no longer includes
+"protected", which is not present in Apple's clang
+
+† *New in 0.52.0* These split visibility attributes are prefered to the plain
```
Comment (QuLogic) [inline docs/markdown/Reference-tables.md]:
```suggestion
† *New in 0.52.0* These split visibility attributes are preferred to the plain
```

File: docs/markdown/snippets/split-visibility-attributes.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+## Splitting of Compiler.get_function_attribute('visibility')
+
+On macOS their is no `protected` visibility, which results in the visbility
```
Comment (QuLogic) [inline docs/markdown/snippets/split-visibility-attributes.md]:
```suggestion
On macOS there is no `protected` visibility, which results in the visbility
```

File: docs/markdown/snippets/split-visibility-attributes.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+check always failing. 0.52.0 introduces two changes to improve this situation:
+
+1. the "visibility" check no longer includes "protected"
+2. a new set of "split" checks are introduced which check for a single
+   attribute instead of all attributes.
+
+These no attributes are
```
Comment (QuLogic) [inline docs/markdown/snippets/split-visibility-attributes.md]:
```suggestion
These new attributes are:
```

File: mesonbuild/compilers/c_function_attributes.py [hunk 4]
```diff
@@ -93,8 +93,11 @@
     'visibility': '''
         int foo_def(void) __attribute__((visibility("default")));
         int foo_hid(void) __attribute__((visibility("hidden")));
-        int foo_int(void) __attribute__((visibility("internal")));
-        int foo_pro(void) __attribute__((visibility("protected")));''',
+        int foo_int(void) __attribute__((visibility("internal")));''',
+    'visibility:default': 'int foo(void) __attribute((visibility("default")))',
```
Comment (QuLogic) [inline mesonbuild/compilers/c_function_attributes.py]: Is this not `__attribute__`?

File: mesonbuild/compilers/c_function_attributes.py [hunk 5]
```diff
@@ -93,8 +93,15 @@
...
         int foo_hid(void) __attribute__((visibility("hidden")));
-        int foo_int(void) __attribute__((visibility("internal")));
-        int foo_pro(void) __attribute__((visibility("protected")));''',
+        int foo_int(void) __attribute__((visibility("internal")));''',
+    'visibility:default':
+        'int foo(void) __attribute__((visibility("default")));',
+    'visibility:hidden':
+        'int foo(void) __attribute__((visibility("hidden")));',
+    'visibility:internal':
+       'int foo(void) __attribute__((visibility("internal")));',
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/c_function_attributes.py]: __[Flake8]__ [E131] continuation line unaligned for hanging indent ([link](https://sider.review/gh/repos/19784232/pulls/5531#issue-4274016))

<sub>posted by Sider</sub>

**Pull request merged**